### ISSUE TITLE
Arcane Trickster cannot select level 1 spells

### ISSUE THREAD

**Issue by nebulans** [BahamutDragon/pcgen#332]
Creating a Rogue and choosing Arcane Trickster as the archetype at level 3, I should get three level 1 spells, two of which must come from the illusion or enchantment spell lists (PHB p98).  These are not offered at all.  Cantrips are offered as expected.

**Comment by BahamutDragon** [comment]
Re-used the updated fighter code provided by @Wemes - quick test shows that is working nicely. Need to verify the Fighter progression is the same for the rogue.

**Comment by Wemes** [comment]
I saw this was coming up and anticipated doing the same thing. I've been taking my time looking at the paladin and with vacation and holidays I haven't finished that yet. The progression should be the same however the rouge has a different spell list than the fighter.

**Comment by BahamutDragon** [comment]
@Wemes I noticed the known spells was off by 1 for the Fighter.

Since Cast ties to the prepared tab, I will use the slot tracker variables to manage the cast. Not ideal, those going multiclass should understand the rules well enough to manage the resources correctly.

**Comment by BahamutDragon** [comment]
@Wemes the progression is working now. Spell list is same as a wizard. Certain spells are restricted to Illusion or Enchantment and a few others are unrestricted. I avoid the special circumstances that will not expand to additional resources easily. Means more work to handle proper support in the future. Eldritch knight and Arcane Trickster are now tied into the spell slot counter.

**Comment by Wemes** [comment]
Is that keeping in mind that the archtype has a set of "unrestricted"
spells that I made separate from the main pool?

**Comment by BahamutDragon** [comment]
No, did not catch that. That would explain the numbering issue. Best to add a comment for those exceptions to make it easy to follow in the future. I will correct those changes and add that comment.

**Comment by Wemes** [comment]
@nebulans I've made some changes to the Rogue class and the Arcane trickster spell casting, and I've created a pull request to merge it. If/when this is done feel free to check that everything is working as the rules specify and let us know if there are any problems. 

As it stands now there is some limits on PCGen enforcing some of the spell swapping restrictions and it's possible for a character to have a few higher level spells than the rules allow. However the class doesn't run into this issue until something like level 19.
